### PR TITLE
Draft a pure Go rawx

### PR DESCRIPTION
**Pull request by jfsmig**
##### SUMMARY
Add an initial (and partial) implementation of a Golang version of the rawx service.

##### ISSUE TYPE
`enhancement`

##### COMPONENT NAME
`rawx`

##### SDS VERSION
`4.3.0`

##### ADDITIONAL INFORMATION
Hopefully:
* PUT/GET/DELETE should be nearly functional

Sadly:
* The current rawx is not subject to the CI functional tests
* ~We do not check `go` is available, during the build process~
* ~No configuration file is read at strartup~
* The log format doesn't follow the standard
* No signal is managed as it is done in other services
* The central configuration is not considered

### CODE REVIEW

File: rawx/filerepo.go [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+	var err error
+	err = setOrHasXattr(r.root, "user.rawx_server.address", url)
+	if err != nil {
+		return err
+	}
+	err = setOrHasXattr(r.root, "user.rawx_server.namespace", ns)
+	if err != nil {
+		return err
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (jfsmig) [reply rawx/filerepo.go]: To be honest, this should even receive a custom test case!

File: rawx/handler_stat.go [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+}
+
+type statHandler struct {
+	rawx *rawxService
+}
+
+// FIXME(jfs): Shouldn't a HEAD return the Content-Length of the GET on the
+// same resource, but without the body?
+func doCheckStats(rr *rawxRequest) {
+	rr.rep.Header().Set("Accept-Ranges", "none")
```
Comment (murlock) [inline rawx/handler_stat.go]: Should be `Accept-Ranges: bytes`
with real `Content-Length`

*note*: `nginx` always close HTTP connection, check https://trac.nginx.org/nginx/ticket/687

File: rawx/main.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+func checkURL(url string) {
+	addr, err := net.ResolveTCPAddr("tcp", url)
+	if err != nil || addr.Port <= 0 {
+		log.Fatalf("%s is not a valid URL", url)
+	}
+}
+
+// TODO(jfs): the pattern doesn't patch the requirement
+func checkNS(ns string) {
+	if ok, _ := regexp.MatchString("[0-9a-zA-Z]+(\\.[0-9a-zA-Z]+)*", ns); !ok {
```
Comment (murlock) [inline rawx/main.go]: Namespace is not `utf-8` ?

File: rawx/main.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+		return false
+	}
+	if addr.Port <= 0 {
+		return false
+	}
+	return true
+}
+
+func checkNamespace(ns string) bool {
+	ok, _ := regexp.MatchString("[0-9a-zA-Z]+(\\.[0-9a-zA-Z]+)*", ns)
```
Comment (murlock) [inline rawx/main.go]: duplicate of regex, `checkNS` should be called here ?

**Pull request merged**